### PR TITLE
Feat/lsdebug query (WIP)

### PR DESCRIPTION
**Pull request by abist**
merging frequently so our branches don't have many conflicts
initial query stuff and cleaned up stuff a little so tslint doesn't complain

### CODE REVIEW

File: extensions/liveshare/src/providers/queryProvider.ts [hunk 1]
```diff
@@ -7,40 +7,125 @@ import * as azdata from 'azdata';
...
-export class QueryFeature {
-	public registerProvider(): vscode.Disposable {
+export class QueryProvider {
+	private _sharedService: SharedService;
+	private _sharedServiceProxy: SharedServiceProxy;
+
+	public constructor(private _isHost: boolean) { }
+
+	public initialize(isHost: boolean, service: any) {
+		if (this._isHost) {
```
Comment (Charles-Gagnon) [inline extensions/liveshare/src/providers/queryProvider.ts]: Same here - isHost isn't used and service could be typed better

File: extensions/liveshare/src/providers/queryProvider.ts [hunk 2]
```diff
@@ -7,40 +7,125 @@ import * as azdata from 'azdata';
...
 
-export class QueryFeature {
-	public registerProvider(): vscode.Disposable {
+export class QueryProvider {
+	private _sharedService: SharedService;
+	private _sharedServiceProxy: SharedServiceProxy;
+
+	public constructor(private _isHost: boolean) { }
+
+	public initialize(isHost: boolean, service: any) {
```
Comment (Charles-Gagnon) [inline extensions/liveshare/src/providers/queryProvider.ts]: Return types on all methods (here and in the other files)

File: extensions/liveshare/src/providers/statusProvider.ts [hunk 3]
```diff
@@ -16,10 +16,9 @@ export class StatusProvider {
 	private _sharedService: SharedService;
 	private _sharedServiceProxy: SharedServiceProxy;
 
-	public constructor(
-		private _isHost: boolean,
-		service: any) {
+	public constructor(private _isHost: boolean) { }
 
+	public initialize(isHost: boolean, service: any) {
```
Comment (Charles-Gagnon) [inline extensions/liveshare/src/providers/statusProvider.ts]: And once more here

**Comment by kburtram** [comment]
@abist looks good to me.  My latest changes are in feat/liveshare-dev.  I'll merge these changes over there so we can have a unified dev branch.  Please create a new branch of there.  We can address @Charles-Gagnon feedback as we prepare for merging to master later this week.

**Comment by kburtram** [comment]
@abist these changes are in `feat/liveshare-dev` now.

**Pull request merged**